parser/src/ast/function.rs

```diff
@@ -0,0 +1,109 @@
+use air_script_core::VariableType;
+
+use super::{Identifier, IntegrityStmt};
+
+#[derive(Debug, Eq, PartialEq)]
+pub struct Function {
+    name: Identifier,
+    arguments: Vec<FunctionArgument>,
+    return_type: Vec<ReturnType>,
+    body: FunctionBody,
+}
+
+impl Function {
+    /// Creates a new function.
+    pub fn new(
+        name: Identifier,
+        arguments: Vec<FunctionArgument>,
+        return_type: Vec<ReturnType>,
+        body: FunctionBody,
+    ) -> Self {
+        Self {
+            name,
+            arguments,
+            return_type,
+            body,
+        }
+    }
+
+    /// Returns the name of the function.
+    pub fn name(&self) -> &str {
+        self.name.name()
+    }
+
+    /// Returns the arguments of the function.
+    pub fn arguments(&self) -> &Vec<FunctionArgument> {
+        &self.arguments
+    }
+
+    /// Returns the return type of the function.
+    pub fn return_type(&self) -> &Vec<ReturnType> {
+        &self.return_type
+    }
+
+    /// Returns the body of the function.
+    pub fn body(&self) -> &FunctionBody {
+        &self.body
+    }
+}
+
+#[derive(Debug, Eq, PartialEq)]
+pub struct FunctionArgument {
+    name: Identifier,
+    argument_type: ArgumentType,
+}
+
+impl FunctionArgument {
+    /// Creates a new function argument.
+    pub fn new(name: Identifier, argument_type: ArgumentType) -> Self {
+        Self {
+            name,
+            argument_type,
+        }
+    }
+
+    /// Returns the name of the function argument.
+    pub fn name(&self) -> &str {
+        self.name.name()
+    }
+
+    /// Returns the type of the function argument.
+    pub fn argument_type(&self) -> &ArgumentType {
+        &self.argument_type
+    }
+}
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub enum ArgumentType {
+    Scalar,
+    Vector(u64),
+    Matrix(u64, u64),
+}
+
+pub type ReturnType = ArgumentType;
```

Comment:
I'm not sure it's useful to have this distinction, and I think it actually confuses things a bit, because it leads to wondering what the difference is between argument types and return types, when actually they're the same. I would prefer to just call it something like `ValueType` instead and use this same type everywhere.